sponsorship_compassion/models/contracts.py

```diff
@@ -748,7 +749,9 @@ def _on_sponsorship_finished(self):
 
         # Update sponsorships on partner
         partners = self.mapped('partner_id') | self.mapped('correspondant_id')
-        partners._compute_has_sponsorships()
+        partners.number_sponsorships -= 1
+        partners.has_sponsorships = partners.number_sponsorships
+        partners.update_church_sponsorships_number(False)

```

Comment:
You cannot update church in this module (must be done in compassion-switzerland)